app/Http/Controllers/BookingController.php

```diff
@@ -2,7 +2,87 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Booking;
+use App\Models\Room;
+use App\Services\JsonResponseService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+
 class BookingController extends Controller
 {
-    //
+    private JsonResponseService $responseService;
+
+    public function __construct(JsonResponseService $responseService)
+    {
+        $this->responseService = $responseService;
+    }
+
+    public function create(Request $request)
+    {
+        $hidden = ['created_at', 'updated_at'];
+        $booking = new Booking();
+
+        $booking->customer = $request->customer;
+        $booking->guests = $request->guests;
+        $booking->start = $request->start;
+        $booking->end = $request->end;
+
+        if ($request->end < $request->start) {
+            return response()->json($this->responseService->getFormat(
+                'Start date must be before the end date'), 400);
+        }
+
+        $clash = Booking::query()->join('booking_room', 'booking_id', '=', 'booking_id')
+            ->where('room_id', $request->room_id)
+            ->where('end', '>=', $request->start)
+            ->where('start', '<=', $request->start)
+            ->exists();
+
+        if ($clash) {
+            return response()->json($this->responseService->getFormat(
+                'Room unavailable for the chosen dates'), 400);
+
+        }
+
+        $clash2 = Room::query()->select('*')
```

Comment:
Same, clash2 is not a good name - what is clashing?

Reply:
Also, using the query builder to select a single room by id is overkill, really you should use use Room::find()